packages/material-ui/src/Rating/Rating.js

```diff
@@ -37,85 +41,148 @@ function roundValueToPrecision(value, precision) {
   return Number(nearest.toFixed(getDecimalPrecision(precision)));
 }
 
-export const styles = (theme) => ({
-  /* Styles applied to the root element. */
-  root: {
-    display: 'inline-flex',
-    // Required to position the pristine input absolutely
-    position: 'relative',
-    fontSize: theme.typography.pxToRem(24),
-    color: '#faaf00',
-    cursor: 'pointer',
-    textAlign: 'left',
-    WebkitTapHighlightColor: 'transparent',
-    '&$disabled': {
-      opacity: theme.palette.action.disabledOpacity,
-      pointerEvents: 'none',
-    },
-    '&$focusVisible $iconActive': {
-      outline: '1px solid #999',
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+
+  return deepmerge(
+    {
+      ...styles[`size${capitalize(styleProps.size)}`],
+      ...(styleProps.readOnly && styles.readOnly),
+      [`& .${ratingClasses.label}`]: styles.label,
+      [`& .${ratingClasses.icon}`]: {
+        ...styles.icon,
+        ...(styleProps.iconEmpty && styles.iconEmpty),
+        ...(styleProps.iconFilled && styles.iconFilled),
+        ...(styleProps.iconHover && styles.iconHover),
+        ...(styleProps.iconFocus && styles.iconFocus),
+        ...(styleProps.iconActive && styles.iconActive),
+        ...(styleProps.decimal && styles.decimal),
+      },
```

Comment:
```suggestion
      [`& .${ratingClasses.icon}`]: styles.icon,,
      [`& .${ratingClasses.iconEmpty }`]: styles.iconEmpty,
      [`& .${ratingClasses.iconFilled}`]: styles.iconFilled,
      [`& .${ratingClasses.iconHover}`]: styles.iconHover,
      [`& .${ratingClasses.iconFocus}`]: styles.iconFocus,
      [`& .${ratingClasses.iconActive}`]: styles.iconActive,
```

Reply:
Ok, I believe this is the last comment :) the `icon*` props are not part of the `styleProps`, as they are state on the icons themselve.